### ISSUE TITLE
Requesting an "ignore tile" feature for the object setter. (i.e. objTilerIgnore)

### ISSUE THREAD

**Issue by sarysa** [MegamixEngine/MegamixEngine#229]
This is a feature request that has many potential use cases. The general idea is when the autotiler checks a specific tile, once it's determined to have an associated collision object but before said object is added, it searches for an object on that tile. For the sake of discussion I'll pretend it's objTilerIgnore. Upon finding objTilerIgnore, it simply moves on in the loop and doesn't apply collision.

The potential use cases for this include, but are not limited to:
- Your standard "false floor", for better or for worse. i.e. Wily/Cossack's Castle in MM4.
- A "slide out" route for a secret room that doesn't give the player an obvious tell that the secret room is there. Guts Man's Asteroid does this in MAGMML2, and I can think of another level that could've been enhanced by this since an enemy was on the route out...but that enemy was also the tell for said room.
- Cinematics that use Mega Man's or an enemy's physics.
- Avoiding potentially undesirable interactions with solids in an above section. (i.e. MM is standing 4 tiles beneath the ceiling, and you don't want him to bump his head, all the while removing the collision on the bottom row in the section above wouldn't compromise anything)

I can understand potential reasons this might be undesirable, like adding to level load times. I hope this is considered.

**Comment by MightyPrinny** [comment]
You can use different layers

**Comment by sarysa** [comment]
True, but as far as convenience goes, adding layers increases users' chance of making errors. Accidentally forgetting you're in the wrong layer and having to redo one or more sections drawn in the wrong layer, just the awkwardness of switching in general...

Plus, it's very common for designers to test the mechanics with filler tiles and then go over them later, so redraws (and the potential errors that come with them) are worth accounting for, in my opinion.

**Comment by sarysa** [comment]
I've gone and made a proof of concept...really for myself as the hour I dumped into this will likely save myself many in avoided mistakes down the road. I can see why there's some hesitation, as the arbitrary instance create order forces moving the process into Begin Step. That in term necessitated recombining collision as is done in objGlobalControl.

There don't appear to be any side effects, and my tests actually resulted in a slight time reduction (not sure why unless instance_create() is the main expense) on my upcoming contest map. (~20 sections with filler tiles so far) I hash all of the "ignore" objects (which are just empty children of prtEntity with 0 gravity) to make searching next to free. In the cleanup phase, ignore objects are destroyed so as not to add to gameplay load.

Anyway, here it is. If it's found to be useful, by all means adapt it. Though I'm guessing most might be hesitant because it's such a fundamental feature. It'll be proven in MAGMML3. ;)

[objSarysa_ATv2.object.zip](https://github.com/MegamixEngine/MegamixEngine/files/2272796/objSarysa_ATv2.object.zip)

**Comment by ginger-mochi** [comment]
Instance create order is not arbitrary. It is deterministic and ordered by the location of the object in the resource tree. Because the collision objects are placed above entities in the resource tree, the autotiler is able to place solids and such before the create event of any entity executes. Furthermore, at room creation time, all instances exist before their create event is run, so it is valid to check for the existence of an object even at create event time. Therefore, there is no need for your refactoring (moving autotiling into the begin step event), and in my opinion it will only make the codebase harder to understand.

The `objTilerIgnore` is a good idea though, but I am concerned that checking for collision with it will be too expensive. Assuming the number of these objects is low, it may be more practical to iterate through such objects and have them remove their underlying tiles from the `objObjectSetter`'s processing queue in advance of processing the tiles. I'm surprised that you experienced speedup, but I can't see how it could be related to this.
